instrumentation/aws-sdk/aws-sdk-2.2/testing/src/main/groovy/io/opentelemetry/instrumentation/awssdk/v2_2/AbstractAws2SqsTracingTest.groovy

```diff
@@ -0,0 +1,180 @@
+import software.amazon.awssdk.core.client.config.ClientOverrideConfiguration
+import software.amazon.awssdk.regions.Region
+import software.amazon.awssdk.services.sqs.SqsClient
+import software.amazon.awssdk.services.sqs.model.CreateQueueRequest
+import software.amazon.awssdk.services.sqs.model.ReceiveMessageRequest
+import software.amazon.awssdk.services.sqs.model.SendMessageRequest
+import spock.lang.Shared
+
+import static io.opentelemetry.api.trace.SpanKind.CLIENT
+import static io.opentelemetry.api.trace.SpanKind.CONSUMER
+import static io.opentelemetry.api.trace.SpanKind.PRODUCER
+import static io.opentelemetry.semconv.trace.attributes.SemanticAttributes.NetTransportValues.IP_TCP
+
+abstract class AbstractAws2SqsTracingTest extends InstrumentationSpecification {
+
+  private static final StaticCredentialsProvider CREDENTIALS_PROVIDER = StaticCredentialsProvider
+    .create(AwsBasicCredentials.create("my-access-key", "my-secret-key"))
+
+  @Shared
+  def sqs
+  @Shared
+  SqsClient client
+  @Shared
+  int sqsPort
+
+  void configureSdkClient(SdkClientBuilder builder) {
+    builder.overrideConfiguration(createOverrideConfigurationBuilder().build())
+  }
+
+  abstract ClientOverrideConfiguration.Builder createOverrideConfigurationBuilder()
+
+  def setupSpec() {
+
+    sqsPort = PortUtils.findOpenPort()
+    sqs = SQSRestServerBuilder.withPort(sqsPort).withInterface("localhost").start()
+    println getClass().name + " SQS server started at: localhost:$sqsPort/"
+
+    def builder = SqsClient.builder()
+    configureSdkClient(builder)
+    client = builder
+      .endpointOverride(new URI("http://localhost:" + sqsPort))
+      .region(Region.AP_NORTHEAST_1)
+      .credentialsProvider(CREDENTIALS_PROVIDER)
+      .build()
+  }
+
+  def cleanupSpec() {
+    if (sqs != null) {
+      sqs.stopAndWait()
+    }
+  }
+
+  def "simple sqs producer-consumer services"() {
+    setup:
+    CreateQueueRequest createQueueRequest = CreateQueueRequest.builder()
+      .queueName("testSdkSqs")
+      .build()
+    client.createQueue(createQueueRequest)
+
+    when:
+    SendMessageRequest sendMessageRequest = SendMessageRequest.builder()
+      .queueUrl("http://localhost:$sqsPort/000000000000/testSdkSqs")
+      .messageBody("{\"type\": \"hello\"}")
+      .build()
+    client.sendMessage(sendMessageRequest)
+    ReceiveMessageRequest receiveMessageRequest = ReceiveMessageRequest.builder()
+      .queueUrl("http://localhost:$sqsPort/000000000000/testSdkSqs")
+      .build()
+    client.receiveMessage(receiveMessageRequest)
+
+    then:
+    assertTraces(3) {
+      trace(0, 1) {
+
+        span(0) {
+          name "Sqs.CreateQueue"
+          kind CLIENT
+          hasNoParent()
+          attributes {
+            "aws.agent" "java-aws-sdk"
+            "aws.queue.name" "testSdkSqs"
+            "aws.requestId" "00000000-0000-0000-0000-000000000000"
+            "rpc.system" "aws-api"
+            "rpc.service" "Sqs"
```

Comment:
Interesting we are using rpc spec here instead of https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/messaging.md. Is there any history on why it does it that way?